src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -913,6 +922,59 @@ child_hdlr(void)
 	context_reset = true;
 }
 
+/* only free the reserved low fds when application exits or encounters error */
+static void
+free_reserved_low_fd(void)
+{
+	int i;
+
+	for (i = 0; i < low_fd_count; i++)
+		libc_close(low_fd_list[i]);
+	low_fd_count = 0;
+}
+
+/* some applications especially bash scripts use specific low fds directly.
+ * It would be safer to avoid using such low fds (fd < DAOS_MIN_FD) in daos.
+ * We consume such low fds before any daos calls and close them only when
+ * application exits or encounters error.
+ */
+
+static int
+consume_low_fd(void)
+{
+	int rc = 0;
+
+	if (atomic_load_relaxed(&daos_inited) == true)
+		return 0;
+
+	D_MUTEX_LOCK(&lock_reserve_fd);
+	low_fd_count              = 0;
+	low_fd_list[low_fd_count] = libc_open("/", O_PATH | O_DIRECTORY);
+	while (1) {
+		if (low_fd_list[low_fd_count] < 0) {
+			DS_ERROR(errno, "failed to reserve a low fd");
+			goto err;
+		} else if (low_fd_list[low_fd_count] >= DAOS_MIN_FD) {
+			libc_close(low_fd_list[low_fd_count]);
+			break;
+		} else {
+			low_fd_count++;
+		}
+		low_fd_list[low_fd_count] = libc_open("/", O_RDONLY);
+	}
+
+	D_MUTEX_UNLOCK(&lock_reserve_fd);
+	return rc;
+
+err:
+	rc = errno;
+	free_reserved_low_fd();
+	low_fd_count = 0;
```

Comment:
Not required now.

Reply:
Could you please provide more information? Thank you!

Reply:
the free_reserved_low_fd() function sets it.